src/Aspire.Hosting.AWS/CDK/Resources/SQSResourceExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using Amazon.CDK.AWS.SQS;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CDK;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Amazon SQS resources to the application model.
+/// </summary>
+public static class SQSResourceExtensions
+{
+
+    private const string QueueUrlOutputName = "QueueUrl";
+
+    /// <summary>
+    /// Adds an Amazon SQS queue.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="props">The properties of the queue.</param>
+    public static IResourceBuilder<IConstructResource<Queue>> AddSQSQueue(this IResourceBuilder<IResourceWithConstruct> builder, string name, IQueueProps? props = null)
```

Comment:
By putting the extension method on `IResourceWithConstruct` the method shows up more then makes sense. For example when I call the `AddDynamoDBTable` to create a DDB table intellisense then shows `AddSQSQueue` as a method off of the table which is really confusing to look like you can add a queue to a table.

Reply:
From a CDK perspective it allowed to add a Queue construct to a DynamoDB Table, but I agree it introduces confusion and false expectations. I'll change this to only accept the `IResourceBuilder<IStackResource>` so it can only be add to a stack. If the user want to do this to a construct, they can use the `AddConstruct` method.